File: 413. Arithmetic Slices/Java/Arithmetic Slices.java
public class Solution {
    public int numberOfArithmeticSlices(int[] A) {
        int result = 0;
        if (A.length < 2) {
            return 0;
        }
        int gap = A[1] - A[0];
        int length = 2;
        for (int i = 2; i < A.length; i++) {
            if (A[i] - A[i - 1] == gap) {
                length++;
            } else {
                if (length >= 3) {
                    result += (1 + length - 3 + 1) * (length - 3 + 1) / 2;
                }
                gap = A[i] - A[i - 1];
                length = 2;
            }
        }
        if (length >= 3) {
            result += (1 + length - 3 + 1) * (length - 3 + 1) / 2;
        }
        return result;
    }
}

File: 413. Arithmetic Slices/Python/Arithmetic Slices.py
class Solution(object):
    def numberOfArithmeticSlices(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        result = 0
        if len(A) < 2:
            return 0
        gap = A[1] - A[0]
        length = 2
        for i in range(2, len(A)):
            if A[i] - A[i - 1] == gap:
                length += 1
            else:
                if length >= 3:
                    result += (1 + length - 3 + 1) * (length - 3 + 1) // 2
                gap = A[i] - A[i - 1]
                length = 2
        if length >= 3:
            result += (1 + length - 3 + 1) * (length - 3 + 1) // 2
        return result
